pkg/openstack/loadbalancer_test.go

```diff
@@ -973,3 +976,281 @@ func TestLbaasV2_updateServiceAnnotations(t *testing.T) {
+			}
+			if tt.expectedErr == "" {
+				assert.NoError(t, err)
+			}
+			assert.Equal(t, got, tt.want)
+		})
+	}
+}
+
+func Test_buildMonitorCreateOpts(t *testing.T) {
+	type testArg struct {
+		lbaas   *LbaasV2
+		svcConf *serviceConfig
+		port    corev1.ServicePort
+	}
+	tests := []struct {
+		name    string
+		testArg testArg
+		want    v2monitors.CreateOpts
+	}{
+		{
+			name: "test for port protocol udp with ovn provider",
+			testArg: testArg{
+				lbaas: &LbaasV2{
+					LoadBalancer{
+						opts: LoadBalancerOpts{
+							LBProvider: "ovn",
+						},
+						lb: &gophercloud.ServiceClient{
+							Endpoint: "/endpoint/",
+						},
+					},
+				},
+				svcConf: &serviceConfig{
+					healthMonitorDelay:          1,
+					healthMonitorTimeout:        1,
+					healthMonitorMaxRetries:     1,
+					healthMonitorMaxRetriesDown: 1,
+					healthCheckNodePort:         2,
+				},
+				port: corev1.ServicePort{
+					Protocol: corev1.ProtocolUDP,
+				},
+			},
+			want: v2monitors.CreateOpts{
+				Name:           "test for port protocol udp with ovn provider",
+				Type:           "UDP-CONNECT",
+				Delay:          1,
+				Timeout:        1,
+				MaxRetries:     1,
+				MaxRetriesDown: 1,
+			},
+		},
+		{
+			name: "using tcp with ovn provider",
+			testArg: testArg{
+				lbaas: &LbaasV2{
+					LoadBalancer{
+						opts: LoadBalancerOpts{
+							LBProvider: "ovn",
+						},
+					},
+				},
+				svcConf: &serviceConfig{
+					healthMonitorDelay:          1,
+					healthMonitorTimeout:        1,
+					healthMonitorMaxRetries:     1,
+					healthMonitorMaxRetriesDown: 1,
+					healthCheckNodePort:         2,
+				},
+				port: corev1.ServicePort{
+					Protocol: corev1.ProtocolTCP,
+				},
+			},
+			want: v2monitors.CreateOpts{
+				Name:           "using tcp with ovn provider",
+				Type:           "TCP",
+				Delay:          1,
+				Timeout:        1,
+				MaxRetries:     1,
+				MaxRetriesDown: 1,
+			},
+		},
+		{
+			name: "using udp with ovn provider",
+			testArg: testArg{
+				lbaas: &LbaasV2{
+					LoadBalancer{
+						opts: LoadBalancerOpts{
+							LBProvider: "ovn",
+						},
+					},
+				},
+				svcConf: &serviceConfig{
+					healthMonitorDelay:          1,
+					healthMonitorTimeout:        1,
+					healthMonitorMaxRetries:     1,
+					healthMonitorMaxRetriesDown: 1,
+					healthCheckNodePort:         2,
+				},
+				port: corev1.ServicePort{
+					Protocol: corev1.ProtocolUDP,
+				},
+			},
+			want: v2monitors.CreateOpts{
+				Name:           "using udp with ovn provider",
+				Type:           "UDP-CONNECT",
+				Delay:          1,
+				Timeout:        1,
+				MaxRetries:     1,
+				MaxRetriesDown: 1,
+			},
+		},
+		{
+			name: "using tcp protocol with not-ovn provider",
+			testArg: testArg{
+				lbaas: &LbaasV2{
+					LoadBalancer{
+						opts: LoadBalancerOpts{
+							LBProvider: "not-ovn",
+						},
+						lb: &gophercloud.ServiceClient{
+							Endpoint: "/endpoint/",
+						},
```

Comment:
Do we even need these? I know `canUseHTTPMonitor()` uses it, but we cannot test that codepath anyway as there's no easy way to mock gophercloud in CPO.